inc/saimacsec.h

```diff
@@ -0,0 +1,1478 @@
+     */
+    SAI_MACSEC_ATTR_SCI_IN_INGRESS_RULES,
+
+    /**
+     * @brief Range of MACSEC rule priority values
+     *
+     * @type sai_u32_range_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_RULE_PRIORITY_RANGE,
+
+    /**
+     * @brief Indicates if 32-bit Packer Number (PN) is supported.
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_PN_32BIT_SUPPORTED,
+
+    /**
+     * @brief Indicates if 64-bit Extended Packer Number (PN) is supported.
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_XPN_64BIT_SUPPORTED,
+
+    /**
+     * @brief Indicates if GCM-AES128 cipher-suite is supported.
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_GCM_AES128_SUPPORTED,
+
+    /**
+     * @brief Indicates if GCM-AES256 cipher-suite is supported.
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_GCM_AES256_SUPPORTED,
+
+    /**
+     * @brief List of supported SecTAG offset values for both ingress parsing
+     * and for egress.
+     *
+     * @type sai_u8_list_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_SECTAG_OFFSETS_SUPPORTED,
+
+    /**
+     * @brief Macsec MTU capability on system side (not including Macsec overhead).
+     *
+     * @type sai_uint16_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_SYSTEM_SIDE_MTU,
+
+    /**
+     * @brief Warm boot is supported for all saimacsec objects.
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_WARM_BOOT_SUPPORTED,
+
+    /**
+     * @brief When false disables creation of saimacsec objects during Phy warm-boot.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_ATTR_WARM_BOOT_ENABLE,
+
+    /**
+     * @brief TPID value used to identify packet C-tag.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 0x8100
+     */
+    SAI_MACSEC_ATTR_CTAG_TPID,
+
+    /**
+     * @brief TPID value used to identify packet S-tag.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 0x88A8
+     */
+    SAI_MACSEC_ATTR_STAG_TPID,
+
+    /**
+     * @brief Global setting of read-clear or read-only for statistics read.
+     * The mode parameter for get_macsec_<foo>_stats_ext should match this.
+     *
+     * @type sai_stats_mode_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_ATTR_STATS_MODE,
+
+    /**
+     * @brief Enables physical bypass of MACsec module. Packets can physically
+     * bypass MACsec module only when there is no macsec_port object
+     * associated with Phy line-side port(s).
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_MACSEC_ATTR_PHYSICAL_BYPASS_ENABLE,
+
+    /**
+     * @brief List of MACsec ports
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_PORT
+     */
+    SAI_MACSEC_ATTR_PORT_LIST,
```

Comment:
this is not necessary as user can get a list of macsec port object and then get port from there. However, I think it will be useful to add SAI_MAC_ATTR_SUPPORTED_PORT_LIST so that user know which ports support mac sec.